### PR TITLE
add certs branch

### PR DESCRIPTION
**Pull request by JiangDonglai98**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /target
 **/.DS_Store
+/certs
```
Comment (Congyuwang) [inline .gitignore]: 末行回车

Comment (Congyuwang) [reply .gitignore]: 见https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

File: gen-certs.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+SCRIPT_PATH=$(dirname "$(realpath $0)")
+certsNum=$1
+echo "当前脚本路径: $SCRIPT_PATH"
+echo "构建：$certsNum 组证书"
```
Comment (Congyuwang) [inline gen-certs.sh]: echo输出改英文

Comment (Congyuwang) [reply gen-certs.sh]: 后面同理

**Review by Congyuwang — CHANGES_REQUESTED** [review]
根据上几个comment修改

File: gen-certs.sh (hunk 2 above)
Comment (JiangDonglai98) [reply gen-certs.sh]: finished

File: gen-certs.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+echo "build：$certsNum groups of certification files"
+
+#cd /root
+mkdir certs
+cd certs
+openssl req -x509 -nodes -new -sha256 -days 1024 -newkey rsa:2048 -keyout RootCA.key -out RootCA.pem -subj "/C=US/CN=DirectCommunication-Root-CA"
+
+openssl x509 -outform pem -in RootCA.pem -out RootCA.crt
+
+# 使用循环创建文件夹并在其中生成文件
```
Comment (Congyuwang) [inline gen-certs.sh]: comment 用英文

File: gen-certs.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+SCRIPT_PATH=$(dirname "$(realpath $0)")
+certsNum=$1
+echo "current path: $SCRIPT_PATH"
+echo "build：$certsNum groups of certification files"
+
+#cd /root
```
Comment (Congyuwang) [inline gen-certs.sh]: 删除所有comment掉的代码

**Review by Congyuwang — CHANGES_REQUESTED** [review]
After some thought, I think we should modify the script design a little bit.

The current design is: `./gen-certs.sh <certsNum>`. And it will generate `direct-node-{0..certsNum}`. This is no good for customized nodes name.

I suggest changing the API to: `./gen-certs.sh <certName1> <certName2> ...` to generate certs for any number of certificates.

If you find shell script inconvenient for this task, you may use `python`.

**Comment by Congyuwang** [comment]
If you decide to use python, you may even try to validate if the domain names:
- no duplicate domain names
- domain names follows standard domain name regex
- 
https://www.geeksforgeeks.org/how-to-validate-a-domain-name-using-regular-expression/

File: gen-certs.sh (hunk 4 above)
Comment (JiangDonglai98) [reply gen-certs.sh]: add standard domain checking and duplicate checking

**Review by Congyuwang — APPROVED** [review]
很棒

**Pull request merged**